md-dao/src/main/java/com/wipro/ats/bdre/md/dao/JobDAO.java

```diff
@@ -355,13 +350,13 @@
                 List<BatchStatus> batchStatuses = new ArrayList<BatchStatus>();
                 batchStatuses.add(processedBatchStatus);
                 batchStatuses.add(newBatchStatus);
-                if (batchStatuses.size() != 0) {
+                if (!batchStatuses.isEmpty()) {
                     Criteria updatingFailedBCQEntryCriteria = session.createCriteria(BatchConsumpQueue.class)
-                            .add(Restrictions.in("batchStatus", batchStatuses)).add(Restrictions.isNotNull("batchByTargetBatchId"));
+                            .add(Restrictions.in("batchStatus", batchStatuses)).add(Restrictions.isNotNull(BATCHBYTARGETBATCHID));
                     for (Object subProcessObject : listOfSubProcessCriteria.list()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)